File: base_module/src/main/java/com/leifeng/base/module/net/RequestManner.java
package com.leifeng.base.module.net;

import io.reactivex.Observable;

public class RequestManner {

    public static <T> void Request(Observable<T> observable, OnRequestListener<T> observer) {
        observable.compose(BaseSchedulers.<T>compose()).subscribe(observer);
    }

    public static <T> void ConvertBeanRequest(Observable<RequestResultBean<T>> observable, OnRequestListener<T> observer) {
        observable.compose(BaseConvertDataSchedulers.<T>compose()).subscribe(observer);
    }

    public static <T> void ConvertDataRequest(Observable<RequestResultBean<T>> observable, OnRequestListener<RequestResultBean<T>> observer) {
        observable.compose(BaseConvertBeanSchedulers.<T>compose()).subscribe(observer);
    }
}

File: base_mvvm_model/src/main/java/com/leifeng/android/base/mvvm/model/net/RequestManner.kt
package com.leifeng.android.base.mvvm.model.net

import io.reactivex.Observable

object RequestManner {

    /**
     * 原始方法
     */
    fun <T> Request(observable: Observable<T>, observer: OnRequestListener<T>) {
        observable.compose(BaseSchedulers.compose()).subscribe(observer)
    }

    /**
     * 转换-返回RequestResultBean
     */
    fun <T> ConvertBeanRequest(observable: Observable<RequestResultBean<T>>, observer: OnRequestListener<RequestResultBean<T>>) {
        observable.compose(BaseConvertBeanSchedulers.compose()).subscribe(observer)
    }

    /**
     * 转换-返回RequestResultBean中data对应的
     */
    fun <T> ConvertDataRequest(observable: Observable<RequestResultBean<T>>, observer: OnRequestListener<T>) {
        observable.compose(BaseConvertDataSchedulers.compose()).subscribe(observer)
    }


}
